tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/annotations/LocalTupleAnnotationTest.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2023, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.ballerinalang.test.annotations;
+
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BString;
+import io.ballerina.runtime.internal.TypeChecker;
+import io.ballerina.runtime.internal.types.BAnnotatableType;
+import io.ballerina.runtime.internal.values.TupleValueImpl;
+import io.ballerina.runtime.internal.values.TypedescValue;
+import org.ballerinalang.test.BCompileUtil;
+import org.ballerinalang.test.BRunUtil;
+import org.ballerinalang.test.CompileResult;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+/**
+ * Class to test annotation evaluation within local tuple types.
+ *
+ * @since 2201.4.0
+ */
+public class LocalTupleAnnotationTest {
+    private CompileResult result;
+
+    @BeforeClass
+    public void setup() {
+        result = BCompileUtil.compile("test-src/annotations/local_tuple_annotation_test.bal");
+        Assert.assertEquals(result.getErrorCount(), 0);
+    }
+
+    @Test(dataProvider = "dataToTestAnnotationsOfLocalTuple")
+    public void testLocalRecordAnnotations(String function) {
+        BRunUtil.invoke(result, function);
+    }
+
+    @Test(description = "Test annotations of tuple members from the type")
+    public void testTupleAnnotations1() {
+        Object returns = BRunUtil.invoke(result, "testTupleMemberAnnotations1", new Object[]{});
+        TupleValueImpl returnTuple = (TupleValueImpl) returns;
+
+        Object annot1 = ((BAnnotatableType) returnTuple.getType()).getAnnotation(StringUtils.fromString("$field$.0"));
+        Object annot2 = ((BAnnotatableType) returnTuple.getType()).getAnnotation(StringUtils.fromString("$field$.1"));
+
+        Assert.assertTrue(annot1 instanceof BMap);
+        Assert.assertTrue(annot2 instanceof BMap);
```

Comment:
We should assert the values also.

Reply:
I believe you can also use `io.ballerina.runtime.api.utils.TypeUtils#getType` and check by tag instead of the instanceof check.

Reply:
Better yet, why not do these assertions also in Ballerina, similar to others?

Reply:
Updated(https://github.com/ballerina-platform/ballerina-lang/pull/39349)